src/Plugin/RulesDataMatcher/MixedLesserThanMatcher.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\RulesDataMatcher\MixedLesserThanMatcher.
+ */
+
+namespace Drupal\rules\Plugin\RulesDataMatcher;
+
+use Drupal\rules\Context\DataMatcherInterface;
+
+/**
+ * Defines a 'lesser than' matcher.
+ *
+ * @RulesDataMatcher(
+ *   id = "rules_mixed_lesser_than",
+ *   label = @Translation("A 'lesser than' matcher."),
```

Comment:
I am a bit lost here, but I answer where I get it:

strict comparison> I suspected there might be some use cases(I'm thinking of edge cases such as string vs int or null/false/""/0 kind of values), but ok I will get rid of it;
AnyEqualsMatcher> just to make sure I understood: are you suggesting to replace StringEqualsMatcher with AnyEqualsMatcher and get rid of the former and of its case sensitive match?
labels> I am not sure I fully get the problem here, should I use the same as in rules_condition_data_is_operator_options? (ie: 'equals', 'is one of', 'is lower than', 'is greater than')